cargo-cyclonedx/src/generator.rs

```diff
@@ -388,44 +390,93 @@ pub enum GeneratorError {
 }
 
 fn top_level_dependencies(
-    member: &PackageId,
+    root: &PackageId,
     packages: &PackageMap,
     resolve: &ResolveMap,
 ) -> (PackageMap, ResolveMap) {
     log::trace!("Adding top-level dependencies to SBOM");
-    let direct_dep_ids = resolve[member].dependencies.as_slice();
+
+    // Only include packages that have dependency kinds other than "Development"
+    let root_node = strip_dev_dependencies(&resolve[root]);
 
     let mut pkg_result = PackageMap::new();
-    pkg_result.insert(member.to_owned(), packages[member].to_owned());
-    for id in direct_dep_ids {
-        pkg_result.insert(id.to_owned(), packages[id].to_owned());
+    // Record the root package, then its direct non-dev dependencies
+    pkg_result.insert(root.to_owned(), packages[root].to_owned());
+    for id in &root_node.dependencies {
+        pkg_result.insert((*id).to_owned(), packages[id].to_owned());
     }
 
     let mut resolve_result = ResolveMap::new();
-    resolve_result.insert(member.to_owned(), resolve[member].clone());
-    for id in direct_dep_ids {
-        // Clear all dependencies, pretend there is only one level
+    for id in &root_node.dependencies {
+        // Clear all depedencies, pretend there is only one level
         let mut node = resolve[id].clone();
         node.deps = Vec::new();
         node.dependencies = Vec::new();
-        resolve_result.insert(id.to_owned(), node);
+        resolve_result.insert((*id).to_owned(), node);
     }
+    // Insert the root node at the end now that we're done iterating over it
+    resolve_result.insert(root.to_owned(), root_node);
```

Comment:
Does this mean we'll add ourselves to our own dependencies or is this (again) only used for the dependency tree later?

Reply:
This is only used for the dependency tree. We do not add the toplevel package to `components`, preserving the original behavior of the tool.